app/scripts/metamask-controller.js

```diff
@@ -3059,11 +3059,11 @@ export default class MetamaskController extends EventEmitter {
           if (chains.length > 0 && !chains.includes(currentChainIdForOrigin)) {
             const networkClientId =
               this.networkController.findNetworkClientIdByChainId(chains[0]);
+            this.networkController.setActiveNetwork(networkClientId);
```

Comment:
[nit] can we add a comment explaining why the `setActiveNetwork` call needs to come before the `setNetworkClientIdForDomain` call

Reply:
https://github.com/MetaMask/metamask-extension/pull/29936/commits/810829033a4b57fa1b45706a7291c03de97869c5